pktfwd/network.go

```diff
@@ -296,35 +307,72 @@ func (c *TTNClient) queueDownlinks() {
 	}
 }
 
-func (c *TTNClient) fetchAccountServerInfo(gatewayID string) error {
-	gw, err := c.account.FindGateway(gatewayID)
+func (c *TTNClient) fetchAccountServerInfo() error {
+	gw, err := c.account.FindGateway(c.runConfig.ID)
 	if err != nil {
 		return errors.Wrap(err, "Account server error")
 	}
 	c.antennaLocation = gw.AntennaLocation
 	c.token = gw.Token.AccessToken
+	c.tokenExpiry = gw.Token.Expiry
 	c.frequencyPlan = gw.FrequencyPlan
+	c.ctx.WithField("TokenExpiry", c.tokenExpiry).Info("Refreshed account server information")
 	return nil
 }
 
+func (c *TTNClient) refreshAccountInfo() error {
+	c.account = account.NewWithKey(c.runConfig.AuthServer, c.runConfig.Key)
+	return c.fetchAccountServerInfo()
+}
+
+func (c *TTNClient) refreshClients() {
+	c.uplinkMutex.Lock()
+	c.routerMutex.Lock()
+	c.connectToStreams(router.NewRouterClientForGateway(router.NewRouterClient(c.currentRouterConn), c.runConfig.ID, c.token), true)
+	c.routerMutex.Unlock()
+	c.uplinkMutex.Unlock()
+	c.downlinkStreamChange <- true
```

Comment:
It sort of is yes. The idea behind this was that gRPC streams are implemented as channels in Go. When waiting for a downlink, the [`select` block](https://github.com/TheThingsNetwork/packet_forwarder/pull/34/files#diff-7b7ee255a9a9b0193a3ed328635f1ecdL293) is stuck on a specific channel - `downlinkStreamChange` signals that the stream needs to be refreshed, and fits well within the `select` block. What do you think?